main.yml

```diff
@@ -4,6 +4,39 @@ write_files:
   - content: |
       - name: prerequisite
         hosts: 127.0.0.1
+        pre_tasks:
+        - name: set false all var
+          set_fact:
+            etcd: false
+            master: false
+            node: false
+          set_fact:
+            etcd: "{{ lookup('env','etcd') }}"
+            master:  "{{ lookup('env','master') }}"
+            node: "{{ lookup('env','node') }}"
+        - name: render consul service
+          set_fact:
+            consul_template_render: | 
+              - name: "{{ lookup('env','name') }}"
+              {% if etcd is sameas true %}
+              - name: "{{ lookup('env','name') }}-etcd"
+                checks:
+                  - args: ["etcdctl", "--ca-file=/etc/ssl/etcd/ssl/ca.pem", "--cert-file=/etc/ssl/etcd/ssl/member-{{ ansible_hostname }}.pem", "--key-file=/etc/ssl/etcd/ssl/member-{{ ansible_hostname }}-key.pem", "--endpoint=https://127.0.0.1:2379", "cluster-health", "|", "grep", "{{ ansible_default_ipv4['address']  }}", "|", "grep", "healthy"]
+                    interval: "10s"
+              {% endif %}
+              {% if master is sameas true %}
+              - name: "{{ lookup('env','name') }}-kube-master"
+                checks:
+                  - args: ["/opt/consul/chek_node.sh"]
```

Comment:
fix typo chek to check

install /opt/consul/check_node.sh before consul configure